### PR TITLE
Resolves: Add native GitHub security and versioning dependency alerts

### LINKED ISSUE #849: Add native GitHub security and versioning dependency alerts
- [Dependabot](https://dependabot.com/) alerts on updates for the project's dated and vulnerable dependencies and automatically creates PRs to easily update them

Pipeline Foundation is **a non-profit initiative** with the sole purpose of giving back to the IT community by assisting OSS projects with DevOps implementations and best practices.

Built with ❤ by [Pipeline Foundation](https://pipeline.foundation).

### PR DESCRIPTION
**Pull request by aleks-ivanov**
- add `dependabot.yml` which automatically enables Dependabot's dependency versioning scanner and dependency update PRs bot by declaring dependency ecosystems and sources in the project. For dependency security vulnerabilities scanner and vulnerable dependency update PRs bot, [enable "Dependabot alerts" and "Dependabot security updates"](https://docs.github.com/en/github/managing-security-vulnerabilities/configuring-dependabot-security-updates)

- use the `target-branch` attribute, if you would like to run Dependabot's scan against a branch other than your default branch (for example if you have a separate development branch)

- should you decide that certain people on your team should take care of the PRs that Dependabot creates, use the two attributes `assignees` and `reviewers` to automatically set personnel respectively.

Resolves #849

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,516 @@
+version: 2
+registries:
+  visual-studio-securitytools:
+    type: nuget-feed
+    url: https://securitytools.pkgs.visualstudio.com/_packaging/Guardian/nuget/v3/index.json
+    # username: my-username
+    # token: ${{ secrets.NUGET_PRIVATE_REG_TOKEN }}
```
Comment (aleks-ivanov) [inline .github/dependabot.yml:6]: To allow Dependabot suggest updates for dependencies in `eng/common/sdl/packages.config`:
- add registry token as GitHub secret with the name `NUGET_PRIVATE_REG_TOKEN` and uncomment

**Comment by Haplois** [comment]
@aleks-ivanov, Thank you for all your work around this; we'll review it and merge. Please let us now when it's done.

**Comment by aleks-ivanov** [comment]
@Haplois our pleasure entirely! 

I would say it is completed now, with the small caveat that we are not 100% sure if the authentication to the private registry requires anything else. We assumed from the URL that it is an Azure auth with a token only. 
If it turns out a usr/pwd combination is needed, replace the `token` attribute with:

```
username: my-usr
password: ${{ secrets.PRIVATE_REG_PWD }}
```

Please don't hesitate to let us know if you have any question about the contribution. 🙂

**Comment by Haplois** [comment]
@aleks-ivanov thank you for this PR, after careful consideration we decided that this is not suitable to our needs, mainly because we have internal tools which tracks this across our organization.

Tools like dependabot might break our compatibility with our other repositories

**Pull request closed without merging**